File: js/examples/watchPlaywright.ts
import { chromium } from 'playwright';
import { CaptchaKrakenSolver } from 'captchakraken';

async function main() {
  const url = process.argv[2] ?? 'https://www.google.com/recaptcha/api2/demo';
  const seconds = Number(process.argv[3] ?? 30);

  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  const solver = new CaptchaKrakenSolver();
  const watcher = solver.watch(page, {
    onSolved: (r) => console.log('✅ solved one:', r.isSolved),
    onError: (e) => console.warn('solve failed:', (e as Error).message),
  });

  await page.goto(url);
  await page.waitForTimeout(seconds * 1000);

  await watcher.stop();
  console.log(`stopped after ${watcher.solves} solve(s)`);
  await browser.close();
}

main().catch((e) => { console.error(e); process.exit(1); });

File: python/examples/watch_playwright.py
import sys

from playwright.sync_api import sync_playwright

from captchakraken import PageSolver

URL = sys.argv[1] if len(sys.argv) > 1 else "https://www.google.com/recaptcha/api2/demo"
SECONDS = int(sys.argv[2]) if len(sys.argv) > 2 else 30


def main() -> int:
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        watcher = PageSolver().watch(
            page,
            on_solved=lambda r: print("✅ solved one:", r.is_solved),
            on_error=lambda e: print("solve failed:", e),
        )

        page.goto(URL)
        watcher.run(timeout_ms=SECONDS * 1000)

        print(f"stopped after {watcher.solves} solve(s)")
        browser.close()
        return 0 if watcher.solves else 1


if __name__ == "__main__":
    raise SystemExit(main())
